src/Features/Core/Portable/CodeRefactorings/CodeRefactoring.cs

```diff
@@ -16,14 +17,14 @@ internal class CodeRefactoring
         /// <summary>
         /// List of possible actions that can be used to transform the code.
         /// </summary>
-        public ImmutableArray<CodeAction> Actions { get; }
+        public ImmutableArray<(CodeAction action, TextSpan? applicableToSpan)> CodeActions { get; }
```

Comment:
so i can see this is the crux of hte change.  You're passing along this additional piece of info.  Can you give me a rundown (or point me to where you've already done this) on what this piece of data will be used for?  What's the intuition behind how it will change the user experience?  Thanks!

Reply:
Long version: https://github.com/dotnet/roslyn/issues/35525#issuecomment-517046330

Short version: AoRN the `CodeRefactoring` & `SuggestedActionsSource` services sort all retrieved (code-fixes, refactorings) by (among others) `ApplicableToSpan`. The `ApplicableToSpan` for (all) Refactorings is, however, set to current selection.

This change enables Refactorings to [register](https://github.com/dotnet/roslyn/pull/37777/files#diff-4b18194545a98c011598321cbfa1dd9fR67) their `ApplicableToSpan` and [flows](https://github.com/dotnet/roslyn/pull/37777/files#diff-2395e5c70cda2d55f3f315c63f3cc59dR789) it to the `SuggestedActionsSource ` to be used for sorting (instead of [statically setting](https://github.com/dotnet/roslyn/pull/37777/files#diff-2395e5c70cda2d55f3f315c63f3cc59dR760) the current selection)

The UX change is that we will be able to order refactorings with respect to how far / close they're to current selection instead of relying on static ordering by an attribute.

E.g. `foo((bar($$), 2)` will show add arg. name (for `bar`) first, add tuple ele. name (for `bar()` second; while `foo(($$2, 2))` will show the one for tuple first and for `foo`'s parameter second.

Reply:
Thanks.  I'll read through that and review later :)

Reply:
Sure, as I said in comment above, there're more ways to route the information so I'm looking for architectural feedback mostly.

Reply:
I'd probably just use a named struct for this.  But overall i have no issue with this :)